rotkehlchen/chain/ethereum/modules/curve/decoder.py

```diff
@@ -285,6 +291,61 @@ def _maybe_enrich_curve_transfers(
             event.counterparty = CPT_CURVE
             event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} from the curve pool {CURVE_Y_DEPOSIT}'  # noqa: E501
             return True
+
+        # enrich deposits made via curve deposits contracts
+        if (
+            event.event_type == HistoryEventType.SPEND and
+            event.event_subtype == HistoryEventSubType.NONE and
+            source_address == transaction.from_address and
+            transaction.input_data.startswith(CURVE_DEPOSIT_ADD_LIQUIDITY_METHOD)
+        ):
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+            event.event_type = HistoryEventType.DEPOSIT
+            event.event_subtype = HistoryEventSubType.DEPOSIT_ASSET
+            event.counterparty = CPT_CURVE
+            event.notes = f'Deposit {event.balance.amount} {crypto_asset.symbol} in curve pool'
+            return True
+
+        if (
+            event.event_type == HistoryEventType.RECEIVE and
+            event.event_subtype == HistoryEventSubType.NONE and
+            to_address == transaction.from_address and
+            transaction.input_data.startswith(CURVE_DEPOSIT_ADD_LIQUIDITY_METHOD)
+        ):
+            crypto_asset = event.asset.resolve_to_crypto_asset()
+            event.event_type = HistoryEventType.RECEIVE
+            event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+            event.counterparty = CPT_CURVE
+            event.notes = f'Receive {event.balance.amount} {crypto_asset.symbol} for depositing in the curve pool'  # noqa: E501
```

Comment:
The pool cannot be known since only transfer txns happen.
## Deposits
-> add token e.g. USDT
<- receive wrapped token e.g. usdnCrv

## Withdrawals
-> return wrapped token e.g. usdnCrv
<- receive one or more tokens deposited e.g. dai, usdt, usdc

**All transfer events**

Reply:
Can't the emitter of the add / remove liquidity event be checked? 
![image](https://user-images.githubusercontent.com/63492346/223452226-c075ed0e-a697-4419-9879-dca8b5ce57e2.png)

(took the screenshot from here https://etherscan.io/tx/0x6c28df56ae4a7f784577273f72402a9b6640024327ee952fdde72c9cfdf08da5#eventlog)